query/src/main/java/org/infinispan/query/impl/massindex/AbstractIndexWorker.java

```diff
@@ -0,0 +1,97 @@
+package org.infinispan.query.impl.massindex;
+
+import org.infinispan.Cache;
+import org.infinispan.commons.util.CloseableIterator;
+import org.infinispan.configuration.cache.StateTransferConfiguration;
+import org.infinispan.container.entries.CacheEntry;
+import org.infinispan.distexec.DistributedCallable;
+import org.infinispan.factories.ComponentRegistry;
+import org.infinispan.filter.AcceptAllKeyValueFilter;
+import org.infinispan.filter.KeyValueFilter;
+import org.infinispan.interceptors.locking.ClusteringDependentLogic;
+import org.infinispan.iteration.impl.LocalEntryRetriever;
+import org.infinispan.marshall.core.MarshalledValue;
+import org.infinispan.metadata.Metadata;
+
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Base class for mass indexer tasks.
+ *
+ * @author gustavonalle
+ * @since 7.1
+ */
+public abstract class AbstractIndexWorker implements DistributedCallable<Void, Void, Void> {
+
+   protected Cache<?, ?> cache;
+   protected final Class<?> entity;
+   protected final String indexName;
+   protected IndexUpdater indexUpdater;
+
+   private ClusteringDependentLogic clusteringDependentLogic;
+   private LocalEntryRetriever entryRetriever;
+
+   public AbstractIndexWorker(Class<?> entity, String indexName) {
+      this.entity = entity;
+      this.indexName = indexName;
+   }
+
+   @Override
+   public void setEnvironment(Cache<Void, Void> cache, Set<Void> inputKeys) {
+      this.cache = cache;
+      this.indexUpdater = new IndexUpdater(cache);
+      ComponentRegistry componentRegistry = cache.getAdvancedCache().getComponentRegistry();
+      StateTransferConfiguration stateTransferCfg = cache.getCacheConfiguration().clustering().stateTransfer();
+      entryRetriever = new LocalEntryRetriever(stateTransferCfg.chunkSize(), stateTransferCfg.timeout(), TimeUnit.MILLISECONDS);
```

Comment:
Are you using LocalEntryRetriever just so that when you are in a DIST cache that it won't hit the remote entries?  If so you can just use a Cache with the CACHE_MODE_LOCAL flag and then retrieve it's retriever using componentRegistry.getComponent(EntryRetriever.class) and you won't have to tie yourself to a specific implementation.